tests/conftest.py

```diff
@@ -47,7 +47,7 @@
 def client():
     global _CLIENT
     if _CLIENT is None:
-        _CLIENT = Client(LocalCluster(n_workers=2))
+        _CLIENT = Client(LocalCluster(n_workers=2, threads_per_worker=1))
```

Comment:
Turns out we get race conditions during worker caching when we allow them to use multiple threads.

Reply:
Should we put a lock around the worker cache instead to handle this? (Or make the worker cache thread local?)

Reply:
Sorry - I am probably wring to call this a "race condition."  For some reason, the worker can end up creating >1 ParquetWriter objects for the same output split.  Everything seems to "work" fine, but the number of output files ends up being wrong.

Not really sure how to add a lock here.  It seems that I can avoid the issue if I add a wait and retry block after the first call to `pq_writer_cache.get` in `get_pq_writer`.  I'll try to figure out the "correct" way to ensure synchronization

Reply:
Okay - I am adding a lock for now, because that seems to work fine.